### DISCUSSION TITLE
How to verify that number (that is string) is greater than 0?

### DISCUSSION THREAD

**Discussion by art-qa** [cypress-io/cypress#22293] [Questions and Help]
**HTML code**

```
<h3 class="class name">
    "Messages"
    " (100)"
    "&lrm;"
</h3>
```

I need to take the number " (100)" and verify that it is greater than 0?

**Comment by jjhelguero** [comment]
You'll have to extract the string number and then parse it into an int to make the number assertion. Here is a working [example](https://github.com/jjhelguero/cypress-examples/blob/main/cypress/integration/stringNumberisGreaterThan0.md)

```js
cy.get("h3")
  .invoke("text")
  .then((text) => {
    const matcher = /\((?<number>\d+)\)/;
    return text.match(matcher)?.groups?.number;
  })
  .then(parseInt)
  .should("be.greaterThan", 0);
```

**Reply by art-qa** [reply]
Thank you for response.

Can you explain please what is happening in this lines?
 
```
const matcher = /\((?<number>\d+)\)/;
return text.match(matcher)?.groups?.number;
```

**Reply by marctaylor700** [reply]
The code starts off by getting the h3 class from cy .
`cy.get("h3")`
Then it invokes text on that element, which is an instance of ReactText .
`.invoke("text")`
The text method takes in a callback function as its only parameter, so when invoked, it will call back with whatever value was returned from invoking text on each individual message.
`  .then((text) => `
Next, we use another regular expression to match any string with "100" followed by one or more digits and return the number of matched strings.
`    const matcher = /\((?<number>\d+)\)/;
    return text.match(matcher)?.groups?.number;` 
This regex uses groups , which allows us to capture substrings within our pattern matching (in this case 100).
We then parseInt() what was returned from our matcher because we want to know how many times 100 appears
` .then(parseInt)`

**Reply by srdjanmutlak** [reply]
Try this also (I added explanations in the notes):

![Try This](https://user-images.githubusercontent.com/84793610/174579279-c096e758-2ab7-43da-acf3-68a72e7b7162.png)
